PubSub/src/Topic.php

```diff
@@ -658,10 +658,17 @@ public function publishBatch(array $messages, array $options = [])
      */
     public function batchPublisher(array $options = [])
     {
+        $config = $this->clientConfig;
+
+        // We don't want to forward the credentials to the BatchPubslisher
+        if (isset($config['credentials'])) {
+            unset($config['credentials']);
```

Comment:
Do we only want to unset this when `credentials ` is a `Credentials` object (as opposed to `string` or `array` credentials)?

Also, I am curious why credentials are failing to serialize... there shouldn't be any `Closures` or `callables` as a part of any `Credentials` object.

Reply:
I could do that, however, we have [this line](https://github.com/googleapis/google-cloud-php/blob/26ce6d7f4954ecb2abb6fc3a719cd784ac70ae77/PubSub/src/PubSubClient.php#L193) in the PubSubClient so I think it's always going to be a CredentialsWrapper object.

Regarding the second part. I am not sure where in the nested structure of the CredentialsWrapper we have a closure which is failing to serialize.

I did check the same thing by forcing the `credentials` to be a path to key file or array contents of the key file. None have the error.